domain/secretbackend/service/service.go

```diff
@@ -0,0 +1,580 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package service
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/juju/clock"
+	"github.com/juju/collections/set"
+	"github.com/juju/errors"
+
+	"github.com/juju/juju/cloud"
+	"github.com/juju/juju/core/changestream"
+	coremodel "github.com/juju/juju/core/model"
+	coresecrets "github.com/juju/juju/core/secrets"
+	"github.com/juju/juju/core/status"
+	"github.com/juju/juju/core/watcher"
+	"github.com/juju/juju/domain"
+	"github.com/juju/juju/domain/secretbackend"
+	secretbackenderrors "github.com/juju/juju/domain/secretbackend/errors"
+	"github.com/juju/juju/environs/cloudspec"
+	internalsecrets "github.com/juju/juju/internal/secrets"
+	"github.com/juju/juju/internal/secrets/provider"
+	"github.com/juju/juju/internal/secrets/provider/juju"
+	"github.com/juju/juju/internal/secrets/provider/kubernetes"
+)
+
+// SecretProviderRegistry is a function that returns a secret backend provider for the given backend type.
+type SecretProviderRegistry func(backendType string) (provider.SecretBackendProvider, error)
```

Comment:
Should we define this func in the secret provider package?

Reply:
This is purely for the service to use but nothing else.
It's an alias of secretBackendRegistry.Provider in `internal/secrets/provider`.
It should live inside here would be better.